PMNS_NSIs.cxx

```diff
@@ -0,0 +1,409 @@
+                double delta_emu, double delta_etau, double delta_mutau
+                       ) 
+{
+
+  SetEta(0,0, eta_ee,     0);
+  SetEta(1,1, eta_mumu,   0);
+  SetEta(2,2, eta_tautau, 0);
+
+  SetEta(0,1, eta_emu,  delta_emu );
+  SetEta(0,2,eta_etau ,  delta_etau);
+  SetEta(1,2, eta_mutau, delta_mutau);
+
+}
+
+//......................................................................
+///
+/// Set both mass-splittings at once.
+///
+/// These are Dm_21 and Dm_32 in eV^2.\n
+/// The corresponding Dm_31 is set in the class attributes.
+///
+/// @param dm21 - The solar mass-splitting Dm_21 
+/// @param dm32 - The atmospheric mass-splitting Dm_32 
+///
+void PMNS_NSIs::SetEta(int flvi, int flvj, double val, double phase){
+
+  if(flvi > flvj){
+    cout << "First argument should be smaller or equal to second argument" << endl;
+    cout << "Setting reverse order (Eta_" << flvj << flvi << "). " << endl;
+    int temp = flvi;
+    flvi = flvj;
+    flvj = temp;
+  }
+  if(flvi<0 || flvi>2 || flvj < flvi || flvj > 2){
+    cout << "Eta_" << flvi << flvj << " not valid for " << fNumNus;
+    cout << " neutrinos. Doing nothing." << endl;
+    return;
+  }
+
+  complexD h = val;  
+
+  if(flvi != flvj) h *= complexD(cos(phase), sin(phase)); 
+
+  bool isSame = (fEta[flvi][flvj] == h);
+  
+  if(!isSame) ClearCache();
+
+  fGotES *= isSame;
+  
+  fEta[flvi][flvj] = h;
+  
+}
+
+//......................................................................
+///
+/// Get any given NSI parameter.
+///
+/// Flavours are:\n
+/// - 0 = nue, 1 = numu, 2 = nutau
+///
+/// Requires that flvi < flvj. Will notify you if input is wrong.
+/// If flvi > flvj, will assume reverse order and swap flvi and flvj.
+///
+/// @param flvi  - The first flavour index
+/// @param flvj  - The second flavour index
+///
+complexD PMNS_NSIs::GetEta(int flvi, int flvj){
+
+  if(flvi > flvj){
+    cout << "First argument should be smaller or equal to second argument" << endl;
+    cout << "Setting reverse order (Eta_" << flvj << flvi << "). " << endl;
+    int temp = flvi;
+    flvi = flvj;
+    flvj = temp;
+  }
+  if(flvi<0 || flvi>2 || flvj < flvi || flvj > 2){
+    cout << "Eta_" << flvi << flvj << " not valid for " << fNumNus;
+    cout << " neutrinos. Returning 0." << endl;
+    return zero;
+  }
+
+  return fEta[flvi][flvj];
+
+}
+
+void PMNS_NSIs::SetLowestMass(double m){
+
+double h=m;
+fM=h;
+
+}
+//......................................................................
+///
+/// Set eta_ee parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
+///
+/// @param a - The value of the real parameter eta_ee
+///
+void PMNS_NSIs::SetEta_ee(double a){ SetEta(0,0, a, 0); }
+
+//......................................................................
+///
+/// Set eta_mumu parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
```

Comment:
I don't think this is true. Where are the flags defined that keep track of what's changed?